blockly-core/core/ui/contract_editor/contract_editor.js

```diff
@@ -501,20 +514,57 @@ Blockly.ContractEditor.prototype.setupUIForBlock_ = function(targetFunctionDefin
   goog.style.showElement(goog.dom.getElement('domain-label'), !isEditingVariable);
   goog.style.showElement(goog.dom.getElement('paramAddButton'), !isEditingVariable);
   goog.style.showElement(goog.dom.getElement('description-area'), !isEditingVariable);
+
+  Blockly.ContractEditor.superClass_.show.call(this);
+};
+
+Blockly.ContractEditor.prototype.setupAfterExampleBlocksAdded_ = function() {
+  var isEditingVariable = this.functionDefinitionBlock.isVariable();
+
   if (isEditingVariable) {
-    this.contractSectionView_.setHeaderVisible(false);
-    this.examplesSectionView_.hideCompletely();
-    this.definitionSectionView_.setHeaderVisible(false);
+    this.setupSectionsForVariable_();
   } else {
-    this.expandAllSections_();
+    this.setupSectionsForContract_(this.autoOpenConfig_);
   }
-  Blockly.ContractEditor.superClass_.show.call(this);
+
+  this.autoOpenConfig_ = null;
 };
 
-Blockly.ContractEditor.prototype.expandAllSections_ = function () {
-  this.contractSectionView_.showHeaderAndExpand();
-  this.examplesSectionView_.showHeaderAndExpand();
-  this.definitionSectionView_.showHeaderAndExpand();
+Blockly.ContractEditor.prototype.setupSectionsForVariable_ = function () {
+  this.contractSectionView_.setHidden(false);
+  this.contractSectionView_.setHeaderVisible(false);
+
+  this.examplesSectionView_.hideCompletely();
+
+  this.definitionSectionView_.setHidden(false);
+  this.definitionSectionView_.setHeaderVisible(false);
+};
+
+Blockly.ContractEditor.prototype.setupSectionsForContract_ = function (autoOpenConfig) {
+  this.allSections_.forEach(function (sectionView, sectionName) {
+    if (autoOpenConfig) {
+      var shouldHighlight = autoOpenConfig[sectionName + HIGHLIGHT_CONFIG_SUFFIX];
+      if (shouldHighlight) {
+        this.setSectionHighlighted(sectionView);
+      }
+      var shouldCollapse = autoOpenConfig[sectionName + COLLAPSE_CONFIG_SUFFIX];
+      sectionView.setCollapsed(shouldCollapse);
+    } else {
+      sectionView.setHighlighted(false);
+      sectionView.setHidden(false);
+      sectionView.setHeaderVisible(true);
+      var previousOpenCollapseState = sectionView.isCollapsed();
+      // Cycle of collapsed callbacks
```

Comment:
Oops, changed that in the last commit but missed pushing it. I removed the first one, but the second was still necessary.

Because we now allow opening the editor in states where you want to hide certain blocks that are in collapsed sections, we need to call the section view's callback which updates the blocks' visibilities to match the collapsed state.

There may be a better place to trigger this, perhaps globally for all of the sections once they're opened